captum/attr/_utils/lrp_rules.py

```diff
@@ -53,6 +47,8 @@ def _backward_hook_input(grad):
                 self.relevance_input[device] = relevance.data
             else:
                 self.relevance_input[device].append(relevance.data)
+
+            grad.replace_out = relevance
```

Comment:
I wonder why are we setting `replace_out` here. It looks like `_create_backward_hook_input` is called for supported linear layers and `backward_hook_activation` is for supported non-linear layers. Is this because we expect that the output of non-linear layer is the input of the linear layer ?

Maybe we can document the logic here ?

https://github.com/pytorch/captum/blob/460a170224a0b64b01af85f23e5005df44a4056e/captum/attr/_core/lrp.py#L320

Reply:
Yes, the issue is that there are 2 tensor backward hooks on the same tensor, but the output doesn't get propagated as input to the next hook automatically, so need this workaround. Will add some comments to explain it better, thanks!